test/e2e/suite/mixed_mode/mixed_mode_test.go

```diff
@@ -0,0 +1,643 @@
+//go:build e2e
+// +build e2e
+
+package mixed_mode
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"math/rand"
+	"strings"
+	"testing"
+	"time"
+
+	ec2v2 "github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2v2types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	smithyTime "github.com/aws/smithy-go/time"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"gopkg.in/yaml.v2"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/remotecommand"
+
+	"github.com/aws/eks-hybrid/test/e2e"
+	"github.com/aws/eks-hybrid/test/e2e/suite"
+)
+
+var (
+	filePath    string
+	suiteConfig *suite.SuiteConfiguration
+
+	// Node selectors as constants
+	cloudNodeSelector  = map[string]string{"node.kubernetes.io/instance-type": "m5.large"}
+	hybridNodeSelector = map[string]string{"eks.amazonaws.com/compute-type": "hybrid"}
+)
+
+func init() {
+	flag.StringVar(&filePath, "filepath", "", "Path to configuration")
+}
+
+func TestMixedModeE2E(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Mixed Mode E2E Suite")
+}
+
+var _ = SynchronizedBeforeSuite(
+	// This function only runs once, on the first process
+	func(ctx context.Context) []byte {
+		suiteConfig := suite.BeforeSuiteCredentialSetup(ctx, filePath)
+
+		test := suite.BeforeVPCTest(ctx, &suiteConfig)
+		credentialProviders := suite.AddClientsToCredentialProviders(suite.CredentialProviders(), test)
+		osProviderList := suite.OSProviderList(credentialProviders)
+		randomOSProvider := osProviderList[rand.Intn(len(osProviderList))]
```

Comment:
does this mean that in one e2e test run, we pick a random OS & cred provider to run the mixed mode connectivity test?